src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -49,7 +50,11 @@ def download_coverage_artifacts(build_task_id):
 
 def get_github_commit(mercurial_commit):
     url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
-    r = requests.get(url % mercurial_commit)
+    while True:
```

Comment:
actually looking at coverrals i remembered seeing a tools that is more DCVS agnostic and works with Mercurial -> [Coveralls-HG](https://pypi.python.org/pypi/Coveralls-HG)

I'm sure similar tool exists for codecov.